oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineImpl.java

```diff
@@ -151,12 +151,19 @@
         } else {
             LOG.debug("Parsing {} statement: {}", language, statement);
         }
+        QueryEngineSettings settings = context.getSettings();
+        if (statement.length() > (settings.getQueryLengthErrorLimit())){
+            LOG.error("Too large query: " + statement);
+            throw new RuntimeException("Query length "+ statement.length() + " is larger than max supported query length: " + settings.getQueryLengthErrorLimit());
```

Comment:
Maybe use ParseException instead of RuntimeException? We could use another exception as well, e.g. IllegalArgumentException, but we use ParseException below as well; I think it's better than to introduce a new exception type.